cmd/tap-fitter/main.go

```diff
@@ -136,7 +137,35 @@ func (p *TapFitterTemplate) generateDockerfile() error {
 }
 
 func (p *TapFitterTemplate) generateDevfile() error {
+	if err := p.validateFlags(); err != nil {
+		return err
+	}
 
-	// TODO: mimic enough cmd.generate.dockerfile to trigger the new alpha.action.generate_devfile based on template fields
-	return nil
+	t, err := template.New("devfile").Parse(devfileTmpl)
+	if err != nil {
+		// The template is hardcoded in the binary, so if
+		// there is a parse error, it was a programmer error.
+		panic(err)
+	}
+	return t.Execute(p.writer, p)
 }
+
+const devfileTmpl = `schemaVersion: 2.2.0
```

Comment:
We can nail down all the fields as we go, but in general I see this divided into two parts:
1. the cobra wrapper, providing CLI interaction support and really nothing else
2. the functional package, where the majority of this project goes.  We could make available to external callers via putting in pkg, or we could limit to internal-use only by putting in internal.